lib/src/widgets/chat_list.dart

```diff
@@ -0,0 +1,131 @@
+  @override
+  Widget build(BuildContext context) {
+    final isBackgroundImage = widget.backgroundImageFromAsset != null ||
+        widget.backgroundImageFromNetwork != null;
+    return Container(
+      height: MediaQuery.of(context).size.height,
+      decoration: BoxDecoration(
+        color: widget.backgroundColor ?? Colors.transparent,
+        image: isBackgroundImage
+            ? DecorationImage(
+                fit: BoxFit.fill,
+                image: widget.backgroundImageFromAsset != null
+                    ? AssetImage(widget.backgroundImageFromAsset!)
+                        as ImageProvider
+                    : NetworkImage(widget.backgroundImageFromNetwork!),
+              )
+            : null,
+      ),
+      child: Stack(
+        children: [
+          SingleChildScrollView(
+            controller: widget.scrollController,
+            child: Column(
+              children: [
+                GroupedListView<Message, String>(
+                  shrinkWrap: true,
+                  elements: widget.messageList,
+                  groupBy: (element) => element.createdAt.getDateFromDateTime,
+                  itemComparator: (message1, message2) =>
+                      message1.message.compareTo(message2.message),
+                  physics: const NeverScrollableScrollPhysics(),
+                  order: widget.order,
+                  sort: widget.sort,
+                  groupSeparatorBuilder: (separator) =>
+                      widget.groupSeparatorBuilder != null
+                          ? widget.groupSeparatorBuilder!(separator)
+                          : ChatGroupHeader(day: DateTime.parse(separator)),
+                  indexedItemBuilder: (context, message, index) => ChatWidget(
+                    message: message,
+                    isCurrentUser: message.sendBy == widget.currentUser,
+                    outgoingChatBubbleColor: widget.outgoingChatBubbleColor,
+                    inComingChatBubbleColor: widget.inComingChatBubbleColor,
+                    chatBubbleTextStyle: widget.chatBubbleTextStyle,
+                    linkStyle: widget.linkStyle,
+                    shareIcon: widget.shareIcon,
+                    onImageViewTap: widget.onImageViewTap,
+                    onShareButtonPressed: widget.onShareButtonPressed,
+                    shareIconBackgroundColor: widget.shareIconBackgroundColor,
+                    heightOfImageMessage: widget.heightOfImageMessage,
+                    widthOfImageMessage: widget.widthOfImageMessage,
+                    inComingChatBubbleBorderRadius:
+                        widget.inComingChatBubbleBorderRadius,
+                    outgoingChatBubbleBorderRadius:
+                        widget.outgoingChatBubbleBorderRadius,
+                    onLongPress: (yCoordinate) => print("OnLongPress"),
+                  ),
+                ),
+                SizedBox(height: Scaffold.of(context).appBarMaxHeight! / 2.5),
```

Comment:
can you please explain the use case of this widget?

Reply:
When textfield added at that time chat-bubble goes to under textfield. So, to give space I added sizedbox. User add textfield with expanded or text chat-bubble does not go under textfield.